go/chat/sender.go

```diff
@@ -119,6 +131,50 @@ func (s *BlockingSender) addPrevPointersToMessage(ctx context.Context, msg chat1
 	return updated, nil
 }
 
+// Check that the {ConvID,ConvTriple,TlfName} of msgToSend matches both the ConvID and an existing message from the questionable ConvID.
+// `convID` is the convID that `msgToSend` will be posted to.
+// `msgReference` is a validated message from `convID`.
+// The misstep that this method checks for is thus: The frontend may post a message while viewing an "untrusted inbox view".
+// That message (msgToSend) will have the header.{TlfName,TlfPublic} set to the user's intention.
+// But the header.Conv.{Tlfid,TopicType,TopicID} and the convID to post to may be erroneously set to a different conversation's values.
+// This method checks that all of those fields match. Using `msgReference` as the validated link from {TlfName,TlfPublic} <-> ConvTriple.
```

Comment:
@mmaxim: @mlsteele and I both asked the same question just now, which is "If the ConvID we're getting from the GUI is potentially bad, why are we getting it at all? Why not get only the TLF name and maybe the topic ID, and then have the service derive everything else?" I wonder if part of the answer here is that we can't derive the shard ID?